docs: improve private module support documentation (#9711)
renovatebot/renovate: docs/usage/private-modules.md

`baseUrl` can be used if you want to only apply the credentials for a nested path within the host, e.g. `https://registry.company.com/nested/path/`.
If the same credentials apply to all paths on a host, then use `hostName` instead, e.g. `registry.company.com`.
Finally, to apply credentials to all hosts within the domain, use `domainName`, e.g. `company.com`.

> You need to pick _only one_ of these and not try to use multiple at the same time, or you will end up with configuration errors.

Author: I think this reads better with emphasis on the _only one_ part.

I'm not clear on what `try not to use multiple` refers to, I shouldn't use multiples of _what_ item/thing/config option? Can you clarify this in the text please?

> You need to pick _only one_ of these and not try to use multiple at the same time, or you will end up with configuration errors.

Reviewer: ```suggestion
You need to pick _only one_ of these and not configure more than one of these fields within the same host rule, otherwise it will error.
```

Renovate applies theses `hostRules` to every HTTP(s) request which is sent, so they are largely independent of any platform or datasource logic.
With `hostRules` in place, private package lookups should all work.

> ## Looking up Release Notes

Author: We mention four times where Renovate needs credentials (including `- Looking up release notes`), so I think it makes sense to use the same words in the heading for this section.

Currently the preferred way to configure `github.com` credentials for self-hosted Renovate is:

> - Create a read-only Personal Access Token (PAT) for a `github.com` account. This can be any GitHub account, it might be better to create a "empty" account just for this purpose.

Author: The general feeling that I get from this document is that we have some "you might like", "it might be good to", "it may even be best" type of sentences. I don't think this helps the document read clearly.

If we have strong opinions on the best way to configure Renovate bot, we should say so clearly in the docs: "This is the best way, don't do it another way". The current wording is a bit too vague for my taste.

As an example, if we think it's better to create a empty/shell GitHub account for the sole purpose of creating the Personal Access Token (PAT), then we should say so clearly.

### gomod

If a `github.com` token is found in `hostRules`, then it is written out to local git config prior to running `go` commands.

> ???CHECK THIS LINE, IS THE COMMAND CORRECT???: The command run is `git config --global url."https://${token}@github.com/".insteadOf "https://github.com/"`.

Author: ```suggestion
The command run is `git config --global url."https://${token}@github.com/".insteadOf"https://github.com/"`.
```

Is that whitespace between `.insteadOf "https://` correct or should it be removed?

> ???CHECK THIS LINE, IS THE COMMAND CORRECT???: The command run is `git config --global url."https://${token}@github.com/".insteadOf "https://github.com/"`.

Reviewer: ```suggestion
The command run is `git config --global url."https://${token}@github.com/".insteadOf "https://github.com/"`.
```

Confirmed in the source code that the whitespace is correct.

Author: Thanks for checking in the source code! 😄
